### PR TITLE
fixed a regression created by PR #2967

### PR DESCRIPTION
**Pull request by duckki**
- Previously, directives applied on operations weren't properly attached to their parent elements.
- Changed the `Operation` class to inherit from `DirectiveTargetElement` class, which attaches directives properly.

Related: https://github.com/apollographql/federation/issues/2961

### CODE REVIEW

File: query-planner-js/src/__tests__/buildPlan.test.ts [hunk 1]
```diff
@@ -8521,4 +8521,44 @@ describe('handles operations with directives', () => {
...
+        test
+      }
+    `;
+
+    const [api, qp] = composeAndCreatePlanner(subgraph1, subgraph2);
+    const op = operationFromDocument(api, query);
+    const queryPlan = qp.buildQueryPlan(op);
+    const fetch_nodes = findFetchNodes(subgraph1.name, queryPlan.node);
+    expect(fetch_nodes).toHaveLength(1);
+    // Note: The query is expected to carry the `@operation` directive.
```
Comment (dariuszkuc) [inline query-planner-js/src/__tests__/buildPlan.test.ts:8557]: i think that's an old comment from previous test?

Comment (duckki) [reply query-planner-js/src/__tests__/buildPlan.test.ts:8557]: Oh, you are right. Let me clean it up.

**Comment by duckki** [comment]
Thanks @dariuszkuc! I'll work with security on the semgrep failures.

**Comment by o0Ignition0o** [comment]
I'm gonna add a changeset entry with 

```
---
"@apollo/query-planner": patch
"@apollo/composition": patch
"@apollo/federation-internals": patch
---
```

So we can kick off a new release

**Pull request merged**